data: https://www.shuwasystem.co.jp/support/7980html/6341.html 

| File Name  |What   |  
|---|---|
|m_area.csv |地域マスタ|  
|m_store.csv|店舗マスタ|
|tbl_order_202004.csv|注文データ 4月分   |
|tbl_order_202005.csv|注文データ 5月分   |
|tbl_order_202006.csv|注文データ 6月分   |

# Nock1 : read some data!

In [40]:
import pandas as pd
m_area = pd.read_csv('MachineLearning/data/ch1/m_area.csv')
m_area

area_cd wide_area narrow_area
0      TK        東京          東京
1      KN       神奈川         神奈川
2      CH        千葉          千葉
3      SA        埼玉          埼玉
4      IB       北関東          茨城
5      TO       北関東          栃木
6      GU       北関東          群馬

In [38]:
m_store = pd.read_csv('MachineLearning/data/ch1/m_store.csv')
m_store

store_id store_name area_cd
0           1        昭島店      TK
1           2      あきる野店      TK
2           3        足立店      TK
3           4       北千住店      TK
4           5        綾瀬店      TK
..        ...        ...     ...
192       193        桐生店      GU
193       194        高崎店      GU
194       195        館林店      GU
195       196        前橋店      GU
196       999       保守担当      TK

[197 rows x 3 columns]

In [9]:
len(m_store)

7

In [41]:
m_area.head()

area_cd wide_area narrow_area
0      TK        東京          東京
1      KN       神奈川         神奈川
2      CH        千葉          千葉
3      SA        埼玉          埼玉
4      IB       北関東          茨城

In [13]:
tbl_order_4 = pd.read_csv('MachineLearning/data/ch1/tbl_order_202004.csv')
tbl_order_4.head()

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0  79339111        49   C26387220         50         67393872   
1  18941733        85   C48773811         26         91834983   
2  56217880        76   C66287421         36         64409634   
3  28447783       190   C41156423         19         73032165   
4  32576156       191   C54568117         71         23281182   

     order_accept_date       delivered_date  takeout_flag  total_amount  \
0  2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1  2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2  2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3  2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4  2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   

   status  
0       1  
1       2  
2       2  
3       2  
4       2

# Nock2: Union Data
union tbl_order_4 and tbl_order_5

In [14]:
tbl_order_5 = pd.read_csv('MachineLearning/data/ch1/tbl_order_202005.csv')

order_all = pd.concat([tbl_order_4, tbl_order_5],ignore_index=True)
order_all

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
474394  47039360       154   C39765066         90         27121351   
474395  61054188        94   C19465010         41         61513298   
474396  45288914        11   C23883838         40         65359139   
474397  41608372       149   C48350551         43         42138977   
474398  40316733       169   C49087342         61         59737067   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
474394  2020-05-31 21:58:58  2020-05-31 22:42:58             1          2363   
474395  2020-05-31 21:58:58  2020-05-31 22:40:58             0          2112   
474396  2020-05-31 21:58:58  2020-05-31 22:38:58             0          3838   
474397  2020-05-31 21:58:58  2020-05-31 22:52:58             0          1899   
474398  2020-05-31 21:58:58  2020-05-31 22:08:58             0          3901   

        status  
0            1  
1            2  
2            2  
3            2  
4            2  
...        ...  
474394       1  
474395       2  
474396       2  
474397       2  
474398       2  

[474399 rows x 10 columns]

In [17]:
#checking union is working by comparing length
len(order_all) == len(tbl_order_4) + len(tbl_order_5)

True

# Nock 3 : list name of files in a folder

In [19]:
import os 
# get current working directory 
current_dir = os.getcwd()
current_dir

'/home/jovyan/work'

In [20]:
os.listdir('MachineLearning/data/ch1')

['.ipynb_checkpoints',
 'm_area.csv',
 'm_store.csv',
 'tbl_order_202004.csv',
 'tbl_order_202005.csv',
 'tbl_order_202006.csv']

In [22]:
# making path of target files with wild card 
tbl_order_file = os.path.join('MachineLearning/data/ch1', 'tbl_order_*.csv')
tbl_order_file

'MachineLearning/data/ch1/tbl_order_*.csv'

In [23]:
import glob 
tbl_order_files = glob.glob(tbl_order_file)
tbl_order_files

['MachineLearning/data/ch1/tbl_order_202004.csv',
 'MachineLearning/data/ch1/tbl_order_202005.csv',
 'MachineLearning/data/ch1/tbl_order_202006.csv']

# Nock4 
複数データをUnion してみる

In [24]:
order_all = pd.DataFrame()
file = tbl_order_files[0]
order_data = pd.read_csv(file)

print(f'{file}:{len(order_data)}')
order_all = pd.concat([order_all, order_data], ignore_index=True)
order_all

MachineLearning/data/ch1/tbl_order_202004.csv:233260


order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
233255  25546308         3   C24617924         46          3700904   
233256  91693481        45   C68997855          6         37852755   
233257  57908119        36   C16123702         81         90372382   
233258   7145625       174   C80733640         75         48265986   
233259  31301192       151   C23758178          9         13847510   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
233255  2020-04-30 21:58:58  2020-04-30 22:19:58             0          2615   
233256  2020-04-30 21:58:58  2020-04-30 22:13:58             0          2732   
233257  2020-04-30 21:58:58  2020-04-30 22:47:58             0          2363   
233258  2020-04-30 21:58:58  2020-04-30 22:32:58             0          2238   
233259  2020-04-30 21:58:58  2020-04-30 22:52:58             0          1900   

        status  
0            1  
1            2  
2            2  
3            2  
4            2  
...        ...  
233255       2  
233256       2  
233257       2  
233258       2  
233259       2  

[233260 rows x 10 columns]

In [25]:
order_all = pd.DataFrame()
for file in tbl_order_files: 
    order_data = pd.read_csv(file)
    print(f'{file}:{len(order_data)}')
    order_all = pd.concat([order_all, order_data],ignore_index=True)

MachineLearning/data/ch1/tbl_order_202004.csv:233260
MachineLearning/data/ch1/tbl_order_202005.csv:241139
MachineLearning/data/ch1/tbl_order_202006.csv:233301


In [26]:
order_all

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
707695  90872494       191   C35992970         46         51884378   
707696  30167637        35   C53126526         88          6295273   
707697  85345862       118   C25099070         32         15733308   
707698  73038887       100   C27421314         84         67608099   
707699  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
707695  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
707696  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
707697  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
707698  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
707699  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status  
0            1  
1            2  
2            2  
3            2  
4            2  
...        ...  
707695       1  
707696       2  
707697       2  
707698       2  
707699       2  

[707700 rows x 10 columns]

# Nock 5
Deta の統計量を確認する

In [27]:
## 欠損値の確認
order_all.isnull().sum()
# 欠損値がない綺麗なデータが返ってくる

order_id             0
store_id             0
customer_id          0
coupon_cd            0
sales_detail_id      0
order_accept_date    0
delivered_date       0
takeout_flag         0
total_amount         0
status               0
dtype: int64

In [28]:
order_all.describe()

order_id       store_id      coupon_cd  sales_detail_id  \
count  7.077000e+05  707700.000000  707700.000000     7.077000e+05   
mean   5.000989e+07     103.934565      49.514363     5.000335e+07   
std    2.889226e+07      86.374451      28.882605     2.885575e+07   
min    2.200000e+01       1.000000       0.000000     4.600000e+01   
25%    2.501247e+07      51.000000      25.000000     2.504502e+07   
50%    4.999655e+07      99.000000      49.000000     5.004506e+07   
75%    7.508014e+07     148.000000      75.000000     7.499707e+07   
max    9.999980e+07     999.000000      99.000000     9.999965e+07   

        takeout_flag   total_amount         status  
count  707700.000000  707700.000000  707700.000000  
mean        0.259864    2960.651814       3.082908  
std         0.438560     954.378771       2.836278  
min         0.000000     698.000000       1.000000  
25%         0.000000    2308.000000       2.000000  
50%         0.000000    2808.000000       2.000000  
75%         1.000000    3617.000000       2.000000  
max         1.000000    5100.000000       9.000000

total_amount field shows that
- average purches per order is 2960 yen
- minimum purches price is 698 yen
- maximum purches price is 5100 yen

In [29]:
# total_amount の項目だけ見る
order_all['total_amount'].describe()

count    707700.000000
mean       2960.651814
std         954.378771
min         698.000000
25%        2308.000000
50%        2808.000000
75%        3617.000000
max        5100.000000
Name: total_amount, dtype: float64

In [32]:
print(order_all["order_accept_date"].min())
print(order_all["order_accept_date"].max())
print(order_all["delivered_date"].min())
print(order_all["delivered_date"].max())

2020-04-01 11:00:00
2020-06-30 21:58:58
2020-04-01 11:10:00
2020-06-30 22:55:56


# Nock 6: 不要なデータを除外する

In [34]:
order_data = order_all.loc[order_all['store_id']!= 999]
order_data

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
707695  90872494       191   C35992970         46         51884378   
707696  30167637        35   C53126526         88          6295273   
707697  85345862       118   C25099070         32         15733308   
707698  73038887       100   C27421314         84         67608099   
707699  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
707695  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
707696  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
707697  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
707698  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
707699  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status  
0            1  
1            2  
2            2  
3            2  
4            2  
...        ...  
707695       1  
707696       2  
707697       2  
707698       2  
707699       2  

[703880 rows x 10 columns]

# Nock 7: Join with Master Data

In [42]:
#print(m_store)
order_data = pd.merge(order_data, m_store, on='store_id', how='left')
order_data

     store_id store_name area_cd
0           1        昭島店      TK
1           2      あきる野店      TK
2           3        足立店      TK
3           4       北千住店      TK
4           5        綾瀬店      TK
..        ...        ...     ...
192       193        桐生店      GU
193       194        高崎店      GU
194       195        館林店      GU
195       196        前橋店      GU
196       999       保守担当      TK

[197 rows x 3 columns]


order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
703875  90872494       191   C35992970         46         51884378   
703876  30167637        35   C53126526         88          6295273   
703877  85345862       118   C25099070         32         15733308   
703878  73038887       100   C27421314         84         67608099   
703879  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
703875  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
703876  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
703877  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
703878  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
703879  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status store_name area_cd  
0            1        浅草店      TK  
1            2        目黒店      TK  
2            2        本郷店      TK  
3            2        栃木店      TO  
4            2       伊勢崎店      GU  
...        ...        ...     ...  
703875       1       伊勢崎店      GU  
703876       2       代々木店      TK  
703877       2        磯子店      KN  
703878       2        中原店      KN  
703879       2        杉並店      TK  

[703880 rows x 12 columns]

In [43]:
order_data = pd.merge(order_data, m_area, on='area_cd', how='left')
order_data

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
703875  90872494       191   C35992970         46         51884378   
703876  30167637        35   C53126526         88          6295273   
703877  85345862       118   C25099070         32         15733308   
703878  73038887       100   C27421314         84         67608099   
703879  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
703875  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
703876  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
703877  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
703878  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
703879  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status store_name area_cd wide_area narrow_area  
0            1        浅草店      TK        東京          東京  
1            2        目黒店      TK        東京          東京  
2            2        本郷店      TK        東京          東京  
3            2        栃木店      TO       北関東          栃木  
4            2       伊勢崎店      GU       北関東          群馬  
...        ...        ...     ...       ...         ...  
703875       1       伊勢崎店      GU       北関東          群馬  
703876       2       代々木店      TK        東京          東京  
703877       2        磯子店      KN       神奈川         神奈川  
703878       2        中原店      KN       神奈川         神奈川  
703879       2        杉並店      TK        東京          東京  

[703880 rows x 14 columns]

# Nock8: Name codes that are not in master data

In [45]:
# in this case, we will name for "takeout_flag" and "status"
# Case: takeout_flag == 0
order_data.loc[order_data['takeout_flag'] == 0, 'takeout_name'] = 'デリバリー'
# Case: takeout_flag == 1 
order_data.loc[order_data['takeout_flag'] == 1, 'takeout_name'] = 'お持ち帰り'
order_data

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
703875  90872494       191   C35992970         46         51884378   
703876  30167637        35   C53126526         88          6295273   
703877  85345862       118   C25099070         32         15733308   
703878  73038887       100   C27421314         84         67608099   
703879  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
703875  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
703876  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
703877  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
703878  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
703879  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status store_name area_cd wide_area narrow_area takeout_name  
0            1        浅草店      TK        東京          東京        お持ち帰り  
1            2        目黒店      TK        東京          東京        デリバリー  
2            2        本郷店      TK        東京          東京        デリバリー  
3            2        栃木店      TO       北関東          栃木        デリバリー  
4            2       伊勢崎店      GU       北関東          群馬        デリバリー  
...        ...        ...     ...       ...         ...          ...  
703875       1       伊勢崎店      GU       北関東          群馬        お持ち帰り  
703876       2       代々木店      TK        東京          東京        デリバリー  
703877       2        磯子店      KN       神奈川         神奈川        デリバリー  
703878       2        中原店      KN       神奈川         神奈川        デリバリー  
703879       2        杉並店      TK        東京          東京        デリバリー  

[703880 rows x 15 columns]

In [47]:
# Case: status == 0
order_data.loc[order_data['status'] == 0, 'status'] = '受付'
# Case: status == 1 
order_data.loc[order_data['status'] == 1, 'status'] = 'お支払い済み'
# Case: status == 2
order_data.loc[order_data['status'] == 2, 'status'] = 'お渡し済み'
# Case: status == 9
order_data.loc[order_data['status'] == 9, 'status'] = 'キャンセル'
order_data

order_id  store_id customer_id  coupon_cd  sales_detail_id  \
0       79339111        49   C26387220         50         67393872   
1       18941733        85   C48773811         26         91834983   
2       56217880        76   C66287421         36         64409634   
3       28447783       190   C41156423         19         73032165   
4       32576156       191   C54568117         71         23281182   
...          ...       ...         ...        ...              ...   
703875  90872494       191   C35992970         46         51884378   
703876  30167637        35   C53126526         88          6295273   
703877  85345862       118   C25099070         32         15733308   
703878  73038887       100   C27421314         84         67608099   
703879  14799648        39   C90614204          5         13300147   

          order_accept_date       delivered_date  takeout_flag  total_amount  \
0       2020-04-01 11:00:00  2020-04-01 11:18:00             1          4144   
1       2020-04-01 11:00:00  2020-04-01 11:22:00             0          2877   
2       2020-04-01 11:00:00  2020-04-01 11:15:00             0          2603   
3       2020-04-01 11:00:00  2020-04-01 11:16:00             0          2732   
4       2020-04-01 11:00:00  2020-04-01 11:53:00             0          2987   
...                     ...                  ...           ...           ...   
703875  2020-06-30 21:58:58  2020-06-30 22:43:58             1          2112   
703876  2020-06-30 21:58:58  2020-06-30 22:36:58             0          4462   
703877  2020-06-30 21:58:58  2020-06-30 22:42:58             0          3865   
703878  2020-06-30 21:58:58  2020-06-30 22:21:58             0          3319   
703879  2020-06-30 21:58:58  2020-06-30 22:38:58             0          2064   

        status store_name area_cd wide_area narrow_area takeout_name  
0       お支払い済み        浅草店      TK        東京          東京        お持ち帰り  
1        お渡し済み        目黒店      TK        東京          東京        デリバリー  
2        お渡し済み        本郷店      TK        東京          東京        デリバリー  
3        お渡し済み        栃木店      TO       北関東          栃木        デリバリー  
4        お渡し済み       伊勢崎店      GU       北関東          群馬        デリバリー  
...        ...        ...     ...       ...         ...          ...  
703875  お支払い済み       伊勢崎店      GU       北関東          群馬        お持ち帰り  
703876   お渡し済み       代々木店      TK        東京          東京        デリバリー  
703877   お渡し済み        磯子店      KN       神奈川         神奈川        デリバリー  
703878   お渡し済み        中原店      KN       神奈川         神奈川        デリバリー  
703879   お渡し済み        杉並店      TK        東京          東京        デリバリー  

[703880 rows x 15 columns]

# Nock 9: 分析基礎テーブルの出力

In [50]:
output_dir = os.path.join('MachineLearning/data/ch1', 'output_data')
os.makedirs(output_dir, exist_ok=True)

In [51]:
output_file = os.path.join(output_dir, 'order_data.csv')
order_data.to_csv(output_file, index=False)

In [52]:
output_file

'MachineLearning/data/ch1/output_data/order_data.csv'